test_cli/test/utils.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import os
+import random
+import subprocess
+import sys
+import tempfile
+import threading
+
+import rclpy
+
+
+def require_environment_variable(name):
+    """Get environment variable or raise if it does not exist."""
+    env = os.getenv(name)
+    if env is None:
+        raise EnvironmentError('Missing environment variable "%s"' % name)
+    return env
+
+
+class HelperCommand:
+    """Execute a command in the background."""
+
+    def __init__(self, command):
+        self._env = dict(os.environ)
+        self._command = copy.deepcopy(command)
+
+        # Execute python files using same python used to start this test
+        if command[0][-3:] == '.py':
+            self._command.insert(0, sys.executable)
+            self._env['PYTHONUNBUFFERED'] = '1'
+
+    def __enter__(self):
+        self._proc = subprocess.Popen(self._command, env=self._env)
+        return self
+
+    def __exit__(self, t, v, tb):
+        self._proc.kill()
+
+
+class NamedTemporaryFile:
```

Comment:
The naming of this custom class is kind of confusing. It is the same as the one from the Python library but it behaves differently (doesn't return a file handle but the name). Therefore I would suggest to rename it to something more "custom" to avoid confusion.

Reply:
`TemporaryFileWithContent` in deb59989c82059923b1fcfa9436476c7e12b9aec